### PR TITLE
Sigma cleanup

### PR DESCRIPTION
**Pull request by jaegeral**
Friday cleanup in Sigma code.

The current linux rules in the repo where basically broken, and given the discussions with @berggren we decided to not ship any sigma rules (yet) but I decided to ship one rule with a sample event file (which is part of this PR) to enable folks to test the functionality overall.

* introduce one sigma rule to show how it works
* modify the sigma mapping

This PR also will change the expected way people should store their rules (for now)

* refactored the analyzer names to only have three types:
** sigma_sample (with the one that is shipped with timesketch)
** sigma_test (for all files that people deveop on their instance)
** sigma_all (for all other sigma rules they have under data/sigma/

That should make it much cleaner and less broken and easier to test by the maintainers as the literally only have to add the shipped jsonl file and hit the run sigma-sample analyzer to run.

### CODE REVIEW

**Comment by kiddinn** [comment]
you'll need to update the branch, first of all.. and then secondly to fix the tests... unit tests are failing:
```
ERROR: Test analyzer.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/mock/mock.py", line 1369, in patched
    return func(*newargs, **newkeywargs)
  File "/timesketch/timesketch/lib/analyzers/sigma_tagger_test.py", line 27, in test_analyzer
    _ = sigma_tagger.LinuxRulesSigmaPlugin(
AttributeError: module 'timesketch.lib.analyzers.sigma_tagger' has no attribute 'LinuxRulesSigmaPlugin'
----------------------------------------------------------------------
Ran 132 tests in 27.629s
```

**Comment by jaegeral** [comment]
To pass all the tests, I adjusted some lines and refactored minor things to increase readability of the code.

File: timesketch/lib/analyzers/sigma_tagger.py [hunk 1]
```diff
@@ -175,19 +174,19 @@ def run(self):
         return output_string
 
 
-class LinuxRulesSigmaPlugin(SigmaPlugin):
+class SampleRulesSigmaPlugin(SigmaPlugin):
```
Comment (berggren) [inline timesketch/lib/analyzers/sigma_tagger.py:177]: Do we need this? Is it for testing? Sample of what? :)
It clutters the UI with this in the available analyzer list.

File: timesketch/lib/analyzers/sigma_tagger.py [hunk 2]
```diff
@@ -175,19 +174,19 @@ def run(self):
...
     """Sigma plugin to run Linux rules."""
 
-    _RULES_PATH = '../../../data/sigma/rules/linux'
+    _RULES_PATH = '../../../data/sigma_sample'
 
-    NAME = 'sigma_linux'
+    NAME = 'sigma_sample'
 
-class WindowsRulesSigmaPlugin(SigmaPlugin):
+class AllRulesSigmaPlugin(SigmaPlugin):
```
Comment (berggren) [inline timesketch/lib/analyzers/sigma_tagger.py:184]: Do we intend to have all Sigma rules under this analyzer? If that is the case, can we call it simply `SigmaPlugin`?

File: timesketch/lib/analyzers/sigma_tagger.py [hunk 3]
```diff
@@ -175,19 +174,19 @@ def run(self):
...
     """Sigma plugin to run Windows rules."""
 
-    _RULES_PATH = '../../../data/sigma/rules/windows'
+    _RULES_PATH = '../../../data/sigma/rules/'
 
-    NAME = 'sigma_windows'
+    NAME = 'sigma_all'
 
 class TestRulesSigmaPlugin(SigmaPlugin):
```
Comment (berggren) [inline timesketch/lib/analyzers/sigma_tagger.py:191]: Same here - do we want this to be visible in the UI?

File: timesketch/lib/analyzers/sigma_tagger.py [hunk 4]
```diff
@@ -104,41 +104,40 @@ def run(self):
...
-                                    rule_file_path, exception))
+                                'Error generating rule in file {0:s}: {1!s}'
+                                .format(rule_file_path, exception))
                             continue
 
                         for sigma_rule in parsed_sigma_rules:
                             try:
                                 simple_counter += 1
-                                # TODO fix that in Sigma hack to get rid of nested stuff
+                                # TODO fix that in Sigma to remove nested stuff
```
Comment (kiddinn) [inline timesketch/lib/analyzers/sigma_tagger.py:118]: what does this mean? `to remove nested stuff` what stuff? what nested

File: timesketch/lib/analyzers/sigma_tagger.py (hunk 1 above)
Comment (kiddinn) [reply timesketch/lib/analyzers/sigma_tagger.py:177]: agree

File: timesketch/lib/analyzers/sigma_tagger.py [hunk 5]
```diff
@@ -196,6 +195,6 @@ class TestRulesSigmaPlugin(SigmaPlugin):
 
     NAME = 'sigma_test'
 
-manager.AnalysisManager.register_analyzer(LinuxRulesSigmaPlugin)
-manager.AnalysisManager.register_analyzer(WindowsRulesSigmaPlugin)
+manager.AnalysisManager.register_analyzer(SampleRulesSigmaPlugin)
+manager.AnalysisManager.register_analyzer(AllRulesSigmaPlugin)
 manager.AnalysisManager.register_analyzer(TestRulesSigmaPlugin)
```
Comment (kiddinn) [inline timesketch/lib/analyzers/sigma_tagger.py:200]: why do we need three here?

File: timesketch/lib/analyzers/sigma_tagger.py (hunk 1 above)
Comment (jaegeral) [reply timesketch/lib/analyzers/sigma_tagger.py:177]: Yeah so the rationale behind is that we have no specific folder for OS, so I assumed it is better to have one for developing rules that would not be pushed to github.

However I see the issue here.
Would it make more sense if we say we have one Analyzer in the UI and it grabs all rules in the folder?

File: timesketch/lib/analyzers/sigma_tagger.py (hunk 4 above)
Comment (jaegeral) [reply timesketch/lib/analyzers/sigma_tagger.py:118]: this nested stuff: https://github.com/google/timesketch/issues/1199#issuecomment-639475885

File: timesketch/lib/analyzers/sigma_tagger.py (hunk 5 above)
Comment (jaegeral) [reply timesketch/lib/analyzers/sigma_tagger.py]: resolved by removing

File: timesketch/lib/analyzers/sigma_tagger.py (hunk 4 above)
Comment (jaegeral) [reply timesketch/lib/analyzers/sigma_tagger.py:118]: (the adjustment of the comment was only to address too long lines)

File: timesketch/lib/analyzers/sigma_tagger.py (hunk 1 above)
Comment (jaegeral) [reply timesketch/lib/analyzers/sigma_tagger.py]: fixed

File: timesketch/lib/analyzers/sigma_tagger.py (hunk 3 above)
Comment (jaegeral) [reply timesketch/lib/analyzers/sigma_tagger.py]: resolved by removing

File: timesketch/lib/analyzers/sigma_tagger.py [hunk 6]
```diff
@@ -107,46 +107,37 @@ def run(self):
...
+                                'Error generating rule in file {0:s}: {1!s}'
+                                .format(rule_file_path, exception))
                             continue
 
                         for sigma_rule in parsed_sigma_rules:
                             try:
                                 simple_counter += 1
-                                # TODO fix that in Sigma hack to get rid of
-                                # nested stuff. Also see:
+                                # TODO fix that in Sigma to remove nested stuff
```
Comment (berggren) [inline timesketch/lib/analyzers/sigma_tagger.py:118]:
```suggestion
                                # TODO Investigate how to handle .keyword fields in Sigma.
```

File: timesketch/lib/analyzers/sigma_tagger.py (hunk 4 above)
Comment (berggren) [reply timesketch/lib/analyzers/sigma_tagger.py:118]: Adjusted the comment.

File: timesketch/lib/analyzers/sigma_tagger.py (hunk 6 above)
Comment (jaegeral) [reply timesketch/lib/analyzers/sigma_tagger.py]: good suggestion, added

**Pull request merged**